### ISSUE TITLE
Facebook messenger error

### ISSUE THREAD

**Issue by ancashoria** [Microsoft/BotBuilder#2126]
Hi,
I'm having trouble with facebook messenger integration. I have setup everything ok and it worked for me the first time I tried it. Then I worked on the bot using the emulator and when I went back to try it on Fb, it doesn't work any more.

I get this error after sending a message to the bot:

```
WARN: Active dialog '*:/help-order' not found in library.
WARN: Dialog '*:/get-started' not found in library.
WARN: Dialog '*:/help-order' not found in library.
/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/promise/lib/done.js:10
      throw err;
      ^
TypeError: Cannot read property 'replyReceived' of undefined
    at Session.routeToActiveDialog (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/botbuilder/lib/Session.js:496:23)
    at UniversalBot.routeMessage.done.recognize.async.each.disambiguateRoute (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/botbuilder/lib/bots/UniversalBot.js:311:37)
    at /Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/botbuilder/lib/bots/UniversalBot.js:317:21
    at _toString (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/async/lib/async.js:52:16)
    at Object.async.forEachOf.async.eachOf (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/async/lib/async.js:236:30)
    at Object.async.forEach.async.each (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/async/lib/async.js:209:22)
    at /Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/botbuilder/lib/bots/UniversalBot.js:296:19
    at UniversalBot.Library.recognize (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/botbuilder/lib/bots/Library.js:57:13)
    at UniversalBot.routeMessage (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/botbuilder/lib/bots/UniversalBot.js:290:14)
    at /Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/botbuilder/lib/bots/UniversalBot.js:284:80

```

I have commented most of my code and the issue still remains. Is it possible that the bot is stuck in a bat state? How would I go about reseting that state?

Another user (that is also an admin of the fb page) tried the bot and I got a different error in this case:
```
Error: Request to 'https://facebook.botframework.com/v3/conversations/1310236079022907-1203161683053941/activities/mid.1485449660047%3A15124fa161' failed: [400] Bad Request
    at Request._callback (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/botbuilder/lib/bots/ChatConnector.js:434:46)
    at Request.init.self.callback (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/request/request.js:186:22)
    at emitTwo (events.js:100:13)
    at Request.emit (events.js:185:7)
    at Request.<anonymous> (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/request/request.js:1081:10)
    at emitOne (events.js:90:13)
    at Request.emit (events.js:182:7)
    at IncomingMessage.<anonymous> (/Users/ancashoria/Workspace/sfappworks/westelm-botframework/node_modules/request/request.js:1001:12)
    at IncomingMessage.g (events.js:273:16)
    at emitNone (events.js:85:20)
```

Any ideas?

**Comment by ancashoria** [comment]
I forgot to mention, the bot works on emulator, skype and telegram, just not on Fb messenger.

**Comment by eanders-MS** [comment]
@Stevenic can you take a look?

**Comment by Stevenic** [comment]
I finally tracked this issue down and will be releasing a patch later today.

**Comment by slahiri** [comment]
Hi @Stevenic - will this fix be published to npm soon?

**Comment by Stevenic** [comment]
Yes... I'm waiting for one more change to get pushed through.

**Comment by Stevenic** [comment]
Just released v3.7.0 with fix
